compat/src/render.js

```diff
@@ -297,7 +296,7 @@ export const __SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED = {
 			useEffect,
 			useId,
 			useImperativeHandle,
-			useInsertionEffect,
+			useInsertionEffect: useLayoutEffect,
```

Comment:
I'd rather not split that alias across two modules, no. Especially not for the demo.

Reply:
It will make the vite dev and build work.

Reply:
And I get that, but the demo is very minor (I don't think we even use it for anything -- frankly, I'd be up for removal as it's a bit of a mess) and shouldn't change source.

The issue is that if we do provide an implementation for `useInsertionEffect`, now there's two separate places that need to be updated which I don't love.

Reply:
> And I get that, but the demo is very minor

I really like it since it very easy to tinker with preact source. Anyway, I will leave you guys decide with PR.

Reply:
I mean, let's find a different solution that doesn't involve changes to our packages 😅 because at this point it points at a bug in vite

Reply:
I certainly see the value, and do think we need to set up something to facilitate that better (I'll be looking into it). But there's a fair number of issues in this approach, which is why I'm fairly apprehensive about encouraging it. Anything that uses the plugin API (outside of this repo, anyways) won't function correctly.

Reply:
We could just do what I did initially and make a `react-hooks` file with these implementations

Reply:
@JoviDeCroock Not necessarily a bug, but a cyclical reference. index -> render, render -> index. I think Microbundle/rollup does throw warnings because of this (but it can handle it alright as it's just bundling). Vite, serving the unbundled individual modules in the browser, falls over (but again, non-representative as no one is running Preact from source in reality, so...)

Reply:
Took another look: while the compat issue can be reasonably fixed by externalizing the React hook implementations to a new module as mentioned, core has a couple cyclical references that I don't think can be untangled without merging modules or doing an internal barrel file sorta thing. I don't think either are a worthwhile approach.

Best solution is to insert `"preact": "../",` into `/demo/package.json` so that it uses built output instead.

Reply:
> no one is running Preact from source in reality

FWIW, I just did (I wanted a no-build workflow that wasn't minified for development, so I can get familiar with the Preact source during dev). Firefox & Chrome both choked on this cyclical reference. This change, plus adding `.js` extensions to the imports, was all I've needed so far to run Preact from source (IMO it's worth the effort).

Reply:
> FWIW, I just did (I wanted a no-build workflow that wasn't minified for development, so I can get familiar with the Preact source during dev).

There certainly are some situations where this works, with that being one of them, however, this breaks expectations of any lib using the plugin API (which at least the first-party ecosystem _heavily, heavily_ relies upon). As such, we don't recommend it in prod as it'll lead to nasty surprises in all likelihood.

Reply:
@rschristian: Good to know -- thank you!